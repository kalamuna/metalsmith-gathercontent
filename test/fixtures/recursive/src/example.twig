<h1>{{gatherContent['about-us'].name}}</h1>
{% for key, item in gatherContent['about-us'].items %}
  <section class="{{key}}">
    <h2>{{ item.name }}</h2>
      {% if item.items %}
        {% for key, component in item.items %}
          <div>
            {{ component.name }}
            {{ dump(component.config)}}
          </div>
        {% endfor %}
      {% endif %}
  </section>
{% endfor %}
