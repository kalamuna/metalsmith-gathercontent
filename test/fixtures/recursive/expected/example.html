<article class="home">

    <h1 class="home--title">Home</h1>

    <ul class="components">
        <li class="section">
            <label></label>
            <h2 class="this-is-a-section-page--title">This is a section page</h2>
            <h3 class="subtitle">
                <p>There will be no content on this page as it is just a section label. ​</p>
            </h3>
            <h4 class="type">section</h4>
            <div class="value"></div>
            <div class="microcopy"></div>

        </li>
    </ul>

    <ul>
        <li>
            <h2 class="--title"></h2>
            <h3 class="header-section">Header section</h3>
            <h4 class="item">item</h4>
            <ul>
                <li>
                    <label>Headline</lavel>
          <h2 class="--title"></h2>
          <h3 class="subtitle"></h3>
          <h4 class="type">text</h4>
          <div class="value"><h1>
    A Good Way to Learn in Life!
</h1></div>
          <div class="microcopy">This should be a H1, and capitalise initials. </div>
        </li>
                <li>
          <label>Sub-headline</lavel>
          <h2 class="--title"></h2>
          <h3 class="subtitle"></h3>
          <h4 class="type">text</h4>
          <div class="value"><h3>
    Since 1967, we've provided a platform for learning and a space for thinking. Welcome, to the Good Education Centre.
</h3></div>
          <div class="microcopy">Make this a H3, and don't capitalise initials. </div>
        </li>
                <li>
          <label>Header image</lavel>
          <h2 class="--title"></h2>
          <h3 class="subtitle"></h3>
          <h4 class="type">files</h4>
          <div class="value"></div>
          <div class="microcopy">Make this a JPG, 1500*800px. No more than 200kb. </div>
        </li>
              </ul>
          </li>
        <li>
      <h2 class="--title"></h2>
      <h3 class="news-item-1">News item 1</h3>
      <h4 class="item">item</h4>
            <ul>
                <li>
          <label></lavel>
          <h2 class="header-component--title">Header component</h2>
          <h3 class="subtitle"><p>This section (Headline, Sub-headline, and Header image) will be the only visible part of this item on the home page. The rest of the fields will be displayed in a window when someone selects "Read more".</p></h3>
          <h4 class="type">section</h4>
          <div class="value"></div>
          <div class="microcopy"></div>
        </li>
                <li>
          <label>Headline</lavel>
          <h2 class="--title"></h2>
          <h3 class="subtitle"></h3>
          <h4 class="type">text</h4>
          <div class="value"><h1>
    The Big Summer Party
</h1></div>
          <div class="microcopy">Make this a H1, and capitalise initials, e.g. "An Exciting Update from the Energy Company Team!". Keep it short and sweet, and use adjectives like "exciting" to make it seem... exciting.  </div>
        </li>
                <li>
          <label>Sub-headline</lavel>
          <h2 class="--title"></h2>
          <h3 class="subtitle"></h3>
          <h4 class="type">text</h4>
          <div class="value"><p>
    If you only go to one party this year, make it this one. We have hats, we have waistcoats, we have all sorts of pastries and cakes. <mark class="inline-comment" data-thread-id="bdda91c6-c347-46ba-9f0c-5b8e98907dcf">You. Only. Live. Once</mark>.
</p></div>
          <div class="microcopy">This sentence (two at the most!) will sit under the headline. It will also appear on the homepage so should be an enticing summary of the post. If you are unsure what to write, try leading with the question that the post will answer. </div>
        </li>
                <li>
          <label>Header image</lavel>
          <h2 class="--title"></h2>
          <h3 class="subtitle"></h3>
          <h4 class="type">files</h4>
          <div class="value"></div>
          <div class="microcopy">This will sit behind the headline as a background. Make sure it's a JPG, or PNG, and at least 1000px wide and 600px tall. Ask the design team to help you if you are unsure about this. They have a library of pre-formatted images you can browse.</div>
        </li>
                <li>
          <label></lavel>
          <h2 class="body-of-news-item--title">Body of news item</h2>
          <h3 class="subtitle"><p>​This is the main section of this item, it will be displayed in a window when someone clicks the "Read more" link.<span class="redactor-invisible-space"></span></p></h3>
          <h4 class="type">section</h4>
          <div class="value"></div>
          <div class="microcopy"></div>
        </li>
                <li>
          <label>Body of post</lavel>
          <h2 class="--title"></h2>
          <h3 class="subtitle"></h3>
          <h4 class="type">text</h4>
          <div class="value"><blockquote>
    <mark class="inline-comment" data-thread-id="c2c8319b-5865-4828-aedb-f566580d14b9"></mark><br>
    <mark class="inline-comment" data-thread-id="c2c8319b-5865-4828-aedb-f566580d14b9">"It was the best party I have ever been to"</mark>
</blockquote>
<p>
    I think you get the point. This is one of the biggest, most summer themed parties you will ever attend. If you get a ticket that is! They do sell fast (usually in minutes).
</p>
<p>
    Tickets go on sale on Thursday the 18th of May, and can be <a href="http://www.gathercontent.com">purchased via this link</a>.
</p>
<p>
    Don't be stupid, come to the Big Summer Party. Live a little!
</p>
<p>
    We'll be sending out an email and an article with full details about the party in the next few weeks. Watch that space!
</p></div>
          <div class="microcopy">You can use lots of quotes in the body and they will be pulled out of the post. This works really well if you have a few people's comments or opinions of the news. 

Another nice thing to do is to find quotes from Twitter (always link back to the Tweet if you do this), so you might consider having a look over there to see if there is anything relevant. </div>
        </li>
                <li>
          <label>News gallery images </lavel>
          <h2 class="--title"></h2>
          <h3 class="subtitle"></h3>
          <h4 class="type">files</h4>
          <div class="value"></div>
          <div class="microcopy">Include up to ten images related to this item of news. You can speak to the news photographer to get these (if you don't already have them). These will be reformatted and compressed in the CMS, so you can upload raw photographs if you like. </div>
        </li>
              </ul>
          </li>
        <li>
      <h2 class="--title"></h2>
      <h3 class="news-item-2">News item 2</h3>
      <h4 class="item">item</h4>
            <ul>
                <li>
          <label></lavel>
          <h2 class="header-component--title">Header component</h2>
          <h3 class="subtitle"><p>This section (Headline, Sub-headline, and Header image) will be the only visible part of this item on the home page. The rest of the fields will be displayed in a window when someone selects "Read more".</p></h3>
          <h4 class="type">section</h4>
          <div class="value"></div>
          <div class="microcopy"></div>
        </li>
                <li>
          <label>Headline</lavel>
          <h2 class="--title"></h2>
          <h3 class="subtitle"></h3>
          <h4 class="type">text</h4>
          <div class="value"><h1>
    Learning To Cook With Only Vegetables
</h1></div>
          <div class="microcopy">Make this a H1, and capitalise initials, e.g. "An Exciting Update from the Energy Company Team!". Keep it short and sweet, and use adjectives like "exciting" to make it seem... exciting.  </div>
        </li>
                <li>
          <label>Sub-headline</lavel>
          <h2 class="--title"></h2>
          <h3 class="subtitle"></h3>
          <h4 class="type">text</h4>
          <div class="value"><p>
    You don't know how to cook. You might think you know how cook... you don't. In this brilliant members workshop, you'll learn how to cook with only veg!
</p></div>
          <div class="microcopy">This sentence (two at the most!) will sit under the headline. It will also appear on the homepage so should be an enticing summary of the post. If you are unsure what to write, try leading with the question that the post will answer. </div>
        </li>
                <li>
          <label>Header image</lavel>
          <h2 class="--title"></h2>
          <h3 class="subtitle"></h3>
          <h4 class="type">files</h4>
          <div class="value"></div>
          <div class="microcopy">This will sit behind the headline as a background. Make sure it's a JPG, or PNG, and at least 1000px wide and 600px tall. Ask the design team to help you if you are unsure about this. They have a library of pre-formatted images you can browse.</div>
        </li>
                <li>
          <label></lavel>
          <h2 class="body-of-news-item--title">Body of news item</h2>
          <h3 class="subtitle"><p>​This is the main section of this item, it will be displayed in a window when someone clicks the "Read more" link.<span class="redactor-invisible-space"></span></p></h3>
          <h4 class="type">section</h4>
          <div class="value"></div>
          <div class="microcopy"></div>
        </li>
                <li>
          <label>Body of post</lavel>
          <h2 class="--title"></h2>
          <h3 class="subtitle"></h3>
          <h4 class="type">text</h4>
          <div class="value"><p>
    This is a chance to really earn your vegetable stripes. Colleen Anderson will be flying in from San Francisco with the sole intention of teaching you how to cook. With only vegetables. No wheat, no meat, no rice, and no spice. You'll be amazed at the flavours which come from this ancient cooking philosophy.
</p>
<p>
    This is a free members program event, so you'll have to be a member if you want to attend. <a href="http://www.gathercontent.com">Learn about TGEC Membership here</a>.
</p>
<p>
    The cooking will only take around 5 hours, and you don't need to bring anything with you in order to join in. Phone Reggie to book your place (<mark class="inline-comment" data-thread-id="f270257a-fd27-4bfa-926b-deedf2eb817b">07906767610909871</mark>).
</p></div>
          <div class="microcopy">You can use lots of quotes in the body and they will be pulled out of the post. This works really well if you have a few people's comments or opinions of the news. 

Another nice thing to do is to find quotes from Twitter (always link back to the Tweet if you do this), so you might consider having a look over there to see if there is anything relevant. </div>
        </li>
                <li>
          <label>News gallery images </lavel>
          <h2 class="--title"></h2>
          <h3 class="subtitle"></h3>
          <h4 class="type">files</h4>
          <div class="value"></div>
          <div class="microcopy">Include up to ten images related to this item of news. You can speak to the news photographer to get these (if you don't already have them). These will be reformatted and compressed in the CMS, so you can upload raw photographs if you like. </div>
        </li>
              </ul>
          </li>
        <li>
      <h2 class="--title"></h2>
      <h3 class="featured-product">Featured product</h3>
      <h4 class="item">item</h4>
            <ul>
                <li>
          <label>Product name</lavel>
          <h2 class="--title"></h2>
          <h3 class="subtitle"></h3>
          <h4 class="type">text</h4>
          <div class="value"><h2>
    Special Wooly Gloves
</h2></div>
          <div class="microcopy">Just use the product's name, and be sure to capitalise initials. E.g. "Space Hat for Aliens"</div>
        </li>
                <li>
          <label>Image of product</lavel>
          <h2 class="--title"></h2>
          <h3 class="subtitle"></h3>
          <h4 class="type">files</h4>
          <div class="value"></div>
          <div class="microcopy">Make this a JPG, 600*400px!</div>
        </li>
                <li>
          <label>Product summary</lavel>
          <h2 class="--title"></h2>
          <h3 class="subtitle"></h3>
          <h4 class="type">text</h4>
          <div class="value"><p>
    These are possibly the most special wooly gloves you could ever hope to wear. Drop by our centre and see for yourself! Just don't blame us if you can't remove them. They are so comfortable, it's impossible to put into content.
</p></div>
          <div class="microcopy">Short and sweet. You may want to use the summary from the product page as a starting point. </div>
        </li>
                <li>
          <label>Link to full product page</lavel>
          <h2 class="--title"></h2>
          <h3 class="subtitle"></h3>
          <h4 class="type">text</h4>
          <div class="value"><p>
    <a href="https://www.youtube.com/watch?v=dQw4w9WgXcQ">www.gathercontent.com/special-wooly-gloves</a>
</p></div>
          <div class="microcopy">This will be added to a button, which reads "Read more about this product"</div>
        </li>
              </ul>
          </li>
      </ul>
  </article>